openmls/src/group/core_group/apply_proposals.rs

```diff
@@ -92,8 +93,13 @@ impl CoreGroup {
             if let Proposal::Update(update_proposal) = queued_proposal.proposal() {
                 // Check if this is our own update.
                 let sender = queued_proposal.sender();
+                // Only members can send update proposals
+                let hash_ref = match sender {
+                    Sender::Member(hash_ref) => hash_ref,
+                    _ => return Err(CoreGroupError::SenderError(SenderError::NotAMember)),
```

Comment:
Do we test for this somewhere already? Or was this error previously done in `as_key_package_ref()`?

Reply:
The latter. I'm really just replacing `as_key_package_ref()` here. I was also wondering if this should become a validation criterion. I filed #788.